### ISSUE TITLE
Data comparison is inconsistent with numpy

### ISSUE THREAD

**Issue by cs8105** [apache/incubator-mxnet#16964]
In [88]: nd.array([0],dtype='int') < 1e-3                                                                                                                                                                                                   
Out[88]: 

[0]
<NDArray 1 @cpu(0)>

In [89]: np.array([0],dtype='int') < 1e-3                                                                                                                                                                                                   
Out[89]: array([ True])

In [90]:

**Comment by haojin2** [comment]
Hi,
The behavior of nd is not guaranteed to be consistent as official NumPy.
If you're looking for NumPy-compatible behaviors, please try out the new NumPy-compatible interface, available in latest nightly versions & soon to be available in 1.6.0 release version:
```
>>> from mxnet import np
>>> np.array([0],dtype='int') < 1e-3
array([False])
```

**Comment by haojin2** [comment]
@ptrendx I think it's a data type issue:
```
>>> from mxnet import np
>>> np.array([0],dtype='float') < 1e-3
array([ True])
```
I guess `1e-3` was casted to integer in the backend...

**Comment by sxjscience** [comment]
I think the root cause would be the same as https://github.com/apache/incubator-mxnet/issues/16653

**Comment by yzhliu** [comment]
@Tommliu is working on this.

**Comment by yzhliu** [comment]
closed by #18277
